src/test/java/net/openhft/chronicle/queue/util/StackSampler.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2016-2020 chronicle.software
+ *
+ *       https://chronicle.software
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package net.openhft.chronicle.queue.util;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.concurrent.locks.LockSupport;
+
+/**
+ * A utility class for sampling the stack traces of a target thread.
+ * This class creates a background daemon thread which periodically
+ * samples the stack trace of the specified thread and stores the
+ * latest snapshot.
+ */
+public class StackSampler {
```

Comment:
Hi Peter is this new or factored from another class? Might be a good candidate for core